editor/editor/mode/visual.js

```diff
@@ -1,13 +1,22 @@
 /**
+ * External dependencies
+ */
+import { findIndex } from 'lodash';
+
+/**
  * Internal dependencies
  */
 import InserterButton from '../../inserter/button';
 
 function Blocks( { blocks, onChange } ) {
-	const onChangeBlock = ( index ) => ( changes ) => {
+	const onChangeBlock = ( uid ) => ( changes ) => {
+		const index = findIndex( blocks, { uid } );
 		const newBlock = {
```

Comment:
Introducing `lodash` inclines to make suggestions 😁  `_.merge` could avoid the nested spread operators:

```js
const onChangeBlock = ( uid ) => ( attributes ) => {
	const index = findIndex( blocks, { uid } );
	const newBlock = merge( {}, blocks[ index ], { attributes } );
	// ...
```